pkg/client/userd/grpc.go

```diff
@@ -52,6 +55,18 @@ func (s *service) logCall(c context.Context, callName string, f func(context.Con
 }
 
 func (s *service) withSession(c context.Context, callName string, f func(context.Context, trafficmgr.Session) error) (err error) {
+	defer func() {
+		name := strcase.ToSnake(callName)
+		traits := []scout.Entry{}
+		if err != nil {
+			name += "_fail"
+			traits = append(traits, scout.Entry{Key: "error", Value: err.Error()})
+		} else {
+			name += "_success"
```

Comment:
This assumption is not true because many (most) errors are reported in the returned message, not as an gRPC `error`. We should in fact clean this up further. Errors here should only be `grpc.status` errors with proper codes.